source/js/ellipsis-elements.js

```diff
@@ -0,0 +1,162 @@
+   * @return {Integer} width
+   */
+  getWidth: function(textLine) {
+    if(textLine.innerText === "") {
+      return 0;
+    }
+  
+    var lineStyles = window.getComputedStyle(textLine);
+
+    if(EllipsisUtil.hasSpecificHeight(lineStyles)) {
+      return parseInt(lineStyles.width);
+    }
+  },
+
+  /***
+   * @summary Returns the width of the object's parent 
+   * @param {Node} textLines: Node with DOM elements that has a string
+   * @return {Integer} parent's width
+   */
+  getParentsWidth: function(textLine) {
+    if(textLine.innerText === "") {
+      return 0;
+    }
+    
+    var lineStyles = window.getComputedStyle(textLine);
+          
+    if(EllipsisUtil.hasSpecificHeight(lineStyles)) {
+      var parentElement = textLine.parentElement;
+      var parentStyles = window.getComputedStyle(parentElement);
+
+      if(parentElement.className !== "multi-teaser-hotel-name") {
+        return parseInt(parentStyles.width);
+      }
+
+      var hotelRating = parentElement.querySelector(".mot_stars");
+      var hotelRatingWidth = window.getComputedStyle(hotelRating).width;
+
+      return parseInt(parentStyles.width) - parseInt(hotelRatingWidth) - parseInt(lineStyles.marginRight);
+    }
+  },
+};
+
+/*** 
+ * @summary Creates a EllipseableObject and pushes it into teaser. 
+ * @param {Node} element Node element.
+ * @param {Integer} numberOfLines One of the model's attributes.
+ * @param {Function} getStyles A self-made method to extract computed Styles.
+ */
+var EllipseableObject = function(element, getStylesFunction, numberOfLines){
+  this.element = element;
+  this.html = this.element.innerHTML;
+  this.fontSize = EllipsisUtil.getFontSize(this.element);
+  this.numberOfLines = numberOfLines;
+  this.getWidth = getStylesFunction;
+
+  ellipseableObjects.push(this);
+};
+
+/***
+ * @summary Applies Ellipse to the Ellispseable element
+ * @description Calculates the character per line and applies the ellipse
+ * @return {Object} EllipseableObject
+ */
+EllipseableObject.prototype.applyEllipse = function() {
+  var charactersPerLine = characterPerLine.calculate(
+    this.fontSize,
+    this.getWidth(this.element)
+  );
+  if (this.element.innerHTML !== "") {
+    this.element.innerHTML = ellipsis(this.html, charactersPerLine * this.numberOfLines, true);
+  }
+  return this;
+};
+
+/***
+ * @description Get's an array and applies the ellipsis to the described elements
+ * @param {Array} initialConfig: Array with configuration for the elements to apply Ellipsis
+ * @return {Array} ellipseableObjects
+ */
+var initializeEllipseableObjects = function(initialConfig) {
+
+
+  var buildEllipseableObjcts = function (builtObjects, config) {
+    var elements = document.querySelectorAll(config.selector);
+
+    if (!errorHandling.checkElements(elements, function() { return true; })) {
+      return builtObjects;
+    }
+
+    elements.forEach(function (element) {
+      var builtObject = config.type === "multiteaser-textline"
```

Comment:
The condition is very long and not easy to read. I suggest to move the EllipseableObject creation into separate methods to make it more readable. The same for the "config.type" checkings.

Reply:
Done